src/hotspot/cpu/ppc/gc/g1/g1BarrierSetAssembler_ppc.cpp

```diff
@@ -108,7 +108,7 @@ void G1BarrierSetAssembler::gen_write_ref_array_post_barrier(MacroAssembler* mas
 }
 
 void G1BarrierSetAssembler::g1_write_barrier_pre(MacroAssembler* masm, DecoratorSet decorators, Register obj, RegisterOrConstant ind_or_offs, Register pre_val,
-                                                 Register tmp1, Register tmp2, bool needs_frame) {
+                                                 Register tmp1, Register tmp2, unsigned int preservation_level) {
```

Comment:
Wouldn't it make sense to use an enum type here? Or is this something that the PPC toolchain doesn't like? Applies to all occurances of similar change all over the place.

Reply:
Hi Roman,
enum type works. We only don't want to use C++11 features like enum class or typed enums because we will probably backport this change to 11u.

Reply:
IOW, can we write MacroAssembler::RuntimeInvocationPreservationLevel instead of unsigned int there or not? I believe this should be possible (even though it'll still only be treated by the compiler as an int)

Reply:
Yes, that should be totally fine.

As a site note: The compiler's behavior isn't well-defined, though. It could be `unsigned int`, `int`, or any other integral type that can represent the enumerators (that is not bigger than `unsigned int`). I'm not a huge fan of _potentially_ using signed integers for enum and/or flag-like parameters, but as an enum _does_ increase the code's readability and ultimately the developer experience by a lot, we should probably use that instead. Typed enums would have been perfect here, but Martin has already pointed out why that isn't an option.

The true reason why I didn't use an enum in the first place, however, is a different one: I originally intended to implement a flag-like mechanism. As it turns out, that is an absolute overkill as the different preservation levels are incremental. Since then, I haven't touched the types, so yeah...

TL;DR: Good catch, will change it.